scylla/src/transport/connection.rs

```diff
@@ -2023,19 +2023,15 @@ mod tests {
                     let futs = (base..base + batch_size).map(|j| {
                         let q = Query::new("INSERT INTO t (p, v) VALUES (?, ?)");
                         let conn = conn.clone();
-                        async move {
-                            conn.query(&q, (j, vec![j as u8; j as usize]), None)
-                                .await
-                                .unwrap()
-                        }
+                        async move { conn.query(&q, (j, vec![j as u8; j as usize]), None).await }
                     });
-                    futures::future::join_all(futs).await;
+                    futures::future::try_join_all(futs).await
```

Comment:
I think this part is OK, actually. If you have a compound future (i.e. a future that polls other futures, like the one returned by `join_all`), panics from the inner futures will be propagated properly. I'm not sure that changes are needed here.

Reply:
Yeah, I guess we can just panic there.
I was scared of panics because I don't fully understand how tokio handles panics, but it looks like it would be ok.

Although I found one more issue here. `Connection::query` returns a `QueryResponse`, which can be `QueryResponse::Error`, so we have to also check that. Fixed in the new version.